c10/core/CPUCachingAllocator.cpp

```diff
@@ -0,0 +1,105 @@
+#include <c10/core/CPUCachingAllocator.h>
+
+namespace c10 {
+
+namespace {
+thread_local CPUCachingAllocator* caching_allocator_ptr{nullptr};
+} // namespace
+
+std::mutex CPUCachingAllocator::mutex_;
+ska::flat_hash_map<void*, size_t> CPUCachingAllocator::allocation_map_;
```

Comment:
Pray that there are never many small allocations of varying sizes? :o) (Would be nice if the allocator's description had a clear statement of what situations it is suitable for use, and what situations it is likely to do poorly in.)